flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendRunTest.java

```diff
@@ -208,6 +209,31 @@ public void testParallelismWithOverflow() throws Exception {
         testFrontend.run(parameters);
     }
 
+    @Test(expected = CliArgsException.class)
+    public void testParallelismWithInvalidValue() throws Exception {
+        String[] parameters = {"-v", "-p", "-2", getTestJarPath()};
+        Configuration configuration = new Configuration();
+        CliFrontend testFrontend =
+                new CliFrontend(configuration, Collections.singletonList(getCli()));
+        testFrontend.run(parameters);
+    }
+
+    @Test
+    public void testParallelismWithNegativeOne() throws Exception {
+        final Configuration configuration = getConfiguration();
+        String[] parameters = {"-v", "-p", "-1", getTestJarPath()};
```

Comment:
I think it's better to replace -1 with ExecutionConfig.PARALLELISM_DEFAULT.
This also applies to other -1